### PR TITLE
Feature/item menu

### PR DESCRIPTION
**Pull request by RootofalleviI**

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorViewHolder.java [hunk 1]
```diff
@@ -1,49 +1,164 @@
...
+    setItemDisplay(favor, context);
+
+    buttonViewOption.setOnClickListener(view -> {
+      popup = new PopupMenu(context, buttonViewOption);
+      popup.inflate(R.menu.item_menu_active);
+
+      // display "commit" option if the favor is requested by the current user.
+      boolean isRequestedByCurrentUser = currentUser.getUid().equals(favor.getRequesterId());
+      if (isRequestedByCurrentUser)
+        popup.getMenu().findItem(R.id.item_menu_commit).setEnabled(false);
```
Comment (huajian1069) [inline app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorViewHolder.java:85]: Personally, I thought it is better to set it invisible.

Comment (huajian1069) [reply app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorViewHolder.java:85]: Also, it is not always reasonable to show "cancel" MenuItem for users, especially when it is posted by others or already archived.

File: app/src/main/java/ch/epfl/favo/util/CommonTools.java [hunk 2]
```diff
@@ -47,7 +50,7 @@ public static void hideToolBar(NonClickableToolbar toolbar) {
   }
 
   public static String convertTime(Date time) {
-    Format format = new SimpleDateFormat("yyyy MM dd HH:mm");
+    @SuppressLint("SimpleDateFormat") Format format = new SimpleDateFormat("yyyy MM dd HH:mm");
```
Comment (mikanikos) [inline app/src/main/java/ch/epfl/favo/util/CommonTools.java:53]: Would it make sense to use a default locale? I'm not sure, just asking

File: app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorViewHolder.java (hunk 1 above)
Comment (mikanikos) [reply app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorViewHolder.java:85]: I agree, I would remove the commit. For the cancel, this should be shown only for favors created by the current user logged in.

**Pull request merged**